### PR TITLE
GO-3802 Implement add chat

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core/create.go [hunk 1]
```diff
@@ -57,7 +62,7 @@ func (mw *Middleware) addChat(cctx context.Context, objectId, objId string) (str
 	}
 
 	chatDetails := &types.Struct{Fields: map[string]*types.Value{}}
-	chatUniqueKey, err := domain.NewUniqueKey(smartblock.SmartBlockTypeChatObject, objId)
+	chatUniqueKey, err := domain.NewUniqueKey(smartblock.SmartBlockTypeChatObject, objectId)
```
Comment (deff7) [inline core/create.go:65]: linter

File: core/create.go [hunk 2]
```diff
@@ -35,14 +35,19 @@ func (mw *Middleware) ObjectCreate(cctx context.Context, req *pb.RpcObjectCreate
 		TemplateId:    req.TemplateId,
 	}
 	id, newDetails, err := creator.CreateObjectUsingObjectUniqueTypeKey(cctx, req.SpaceId, req.ObjectTypeUniqueKey, createReq)
-
+	if err != nil {
+		return response(pb.RpcObjectCreateResponseError_UNKNOWN_ERROR, "", nil, err)
+	}
+	if req.WithChat {
+		_, err = mw.addChat(cctx, id)
```
Comment (deff7) [inline core/create.go:42]: move error check close to error assignment

File: core/block/object/objectcreator/chat.go [hunk 3]
```diff
@@ -4,28 +4,48 @@ import (
...
+func (s *service) createChatDerived(ctx context.Context, space clientspace.Space, details *types.Struct) (string, *types.Struct, error) {
 	uniqueKey := pbtypes.GetString(details, bundle.RelationKeyUniqueKey.String())
 	key, err := domain.UnmarshalUniqueKey(uniqueKey)
 	if err != nil {
 		return "", nil, fmt.Errorf("unmarshal unique key: %w", err)
 	}
 
 	createState := state.NewDocWithUniqueKey("", nil, key).(*state.State)
-	details.Fields[bundle.RelationKeyLayout.String()] = pbtypes.String(model.ObjectType_chat.String())
+	details.Fields[bundle.RelationKeyLayout.String()] = pbtypes.String(model.ObjectType_chatDerived.String())
```
Comment (deff7) [inline core/block/object/objectcreator/chat.go:45]: are there two layout for chats?

File: core/block/object/objectcreator/chat.go [hunk 4]
```diff
@@ -4,28 +4,48 @@ import (
...
 	"github.com/anyproto/anytype-heart/core/domain"
 	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	smartblock2 "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
 	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
 	"github.com/anyproto/anytype-heart/space/clientspace"
 	"github.com/anyproto/anytype-heart/util/pbtypes"
 )
 
 func (s *service) createChat(ctx context.Context, space clientspace.Space, details *types.Struct) (string, *types.Struct, error) {
+	key, err := domain.NewUniqueKey(smartblock2.SmartBlockTypeChatDerivedObject, bson.NewObjectId().Hex())
```
Comment (deff7) [inline core/block/object/objectcreator/chat.go:20]: derivation makes no sense with random ids

File: core/block/object/objectcreator/chat.go (hunk 3 above)
Comment (fat-fellow) [reply core/block/object/objectcreator/chat.go:45]: yes, i have understood it like that

File: core/block/object/objectcreator/chat.go (hunk 4 above)
Comment (fat-fellow) [reply core/block/object/objectcreator/chat.go:20]: that was typo

**Review by deff7 — APPROVED** [review]

**Pull request merged**